qpmodel/optimizer.cs

```diff
@@ -50,8 +51,61 @@ public class Property { }
     public class PhysicProperty : Property
```

Comment:
See previous comment, you may want to have SortOrderProperty/DistributionProperty etc derived from Property? For clear naming.